### PR TITLE
Add a new metric and a new alert for not completed installation

### PR DESCRIPTION
**Pull request by nunnatsa**
This PR adds a new Prometheus metric `kubevirt_hco_hyperconverged_cr_exists`. HCO set this metric to 1 if the HyperConverged CR is exist, or 0 if not.

The PR also adds a new alert `KubevirtHyperconvergedClusterOperatorInstallationNotCompletedAlert`. This alert is triggered when the new `kubevirt_hco_hyperconverged_cr_exists` metric is not 1 (if the HyperConverged CR does not exist).

The meaning of the new alert is that the installation is not completed, and the user should create the HyperConverged CR to complete the installation.

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly
  
**Release note**:

```release-note
New metric: kubevirt_hco_hyperconverged_cr_exists - 1 if the HyperConverged CR exists, 0 if not

New alert KubevirtHyperconvergedClusterOperatorInstallationNotCompletedAlert if the kubevirt_hco_hyperconverged_cr_exists metric is not 1.
```

### CODE REVIEW

File: controllers/operands/monitoring.go [hunk 1]
```diff
@@ -241,6 +243,20 @@ func NewPrometheusRuleSpec() *monitoringv1.PrometheusRuleSpec {
 						componentAlertLabelKey: componentAlertLabelValue,
 					},
 				},
+				{
+					Alert: installationNotCompletedAlert,
+					Expr:  intstr.FromString("(1 - kubevirt_hco_hyperconverged_cr_exists) > 0"),
+					Annotations: map[string]string{
+						"description": "the installation was not completed; the HyperConverged custom resource is missing.",
```
Comment (sradco) [inline controllers/operands/monitoring.go:250]: @nunnatsa perhaps, 
In order to complete the installation of the Hyperconverged Cluster Operator you should create the HyperConverged custom resource.

On d/s runbooks we can rename to OpenShift Virtualization.

File: controllers/operands/monitoring.go [hunk 2]
```diff
@@ -241,6 +243,20 @@ func NewPrometheusRuleSpec() *monitoringv1.PrometheusRuleSpec {
...
+						"summary":     "the installation was not completed; the HyperConverged custom resource is missing.",
```
Comment (sradco) [inline controllers/operands/monitoring.go:251]: @nunnatsa perhaps
The HyperConverged custom resource is missing.

File: pkg/metrics/metrics.go [hunk 3]
```diff
@@ -12,6 +12,13 @@ import (
 const (
 	counterLabelCompName = "component_name"
 	counterLabelAnnName  = "annotation_name"
+
+	HCOMetricOverwrittenModifications = "overwrittenModifications"
+	HCOMetricUnsafeModifications      = "unsafeModifications"
+	HCOMetricHyperConvergedExists     = "HyperConvergedCRExists"
+
+	HyperConvergedExists   = float64(1)
+	HyperConvergedNotExist = float64(0)
```
Comment (orenc1) [inline pkg/metrics/metrics.go:21]: nit:
`HyperConvergedNotExist` --> `HyperConvergedNotExists`

**Comment by nunnatsa** [comment]
hco-e2e-upgrade-prev-index-sno-azure lane passed

/override ci/prow/hco-e2e-upgrade-prev-index-sno-aws

**Pull request merged**